metadata-dao-impl/kafka-producer/src/main/java/com/linkedin/metadata/dao/producer/KafkaHealthChecker.java

```diff
@@ -0,0 +1,67 @@
+package com.linkedin.metadata.dao.producer;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.producer.Callback;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Slf4j
+@EnableScheduling
+@Component
+public class KafkaHealthChecker {
+
+    @Value("${kafka.producer.requestTimeout}")
+    private long kafkaProducerRequestTimeout;
+
+    @Value("${kafka.producer.backoffTimeout}")
+    private long kafkaProducerBackOffTimeout;
+
+    private static long lastMoment = 0L;
+    private Set<Long> messagesInProgress = new HashSet<>();
+
+    private synchronized long getNextUniqueMoment() {
+        long moment = System.currentTimeMillis();
+        lastMoment = moment != lastMoment ? moment : ++lastMoment;
+        return lastMoment;
+    }
+
+    public Callback getKafkaCallBack(String eventType, String entityDesc) {
+        long moment = getNextUniqueMoment();
+        sendMessageStarted(moment);
+        return (metadata, e) -> {
+            sendMessageEnded(moment);
+            if (e != null) {
+                log.error(String.format("Failed to emit %s for entity %s", eventType, entityDesc), e);
```

Comment:
It might be a good idea to capture a metric here for the count of failed messages. This could then be used by prometheus or other monitoring solutions for alerting. It would be one line addition here, similar to [here](https://github.com/datahub-project/datahub/blob/9434afc78aa54b95b69478fafe39baec1c946136/metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/MetadataChangeLogProcessor.java#L66) and look like `MetricUtils.counter(this.getClass() + "." + eventType, "producer_failed_count").inc();`

Reply:
Might need some logic to replace spaces in the eventType with underscores it looks like.